util/deephash/deephash.go

```diff
@@ -106,9 +106,18 @@ func (s1 *Sum) xor(s2 Sum) {
 }
 
 func (s Sum) String() string {
+	// Note: if we change this, keep in sync with AppendSum
 	return hex.EncodeToString(s.sum[:])
 }
 
+// AppendSum appends the string encoding of this sum (as returned by the String
+// method) to the provided byte slice and returns the extended buffer.
+func (s Sum) AppendSum(b []byte) []byte {
```

Comment:
The `netip` package uses the name `AppendTo` for a method of the same signature. Should we do so as well here? The "sum" part is a little redundant given that the type is `Sum`.

Reply:
SGTM; changed